tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts

```diff
@@ -95,7 +95,12 @@ export class CardFobControllerComponent {
   }
 
   startDrag(fob: Fob, affordance: TimeSelectionAffordance, event: MouseEvent) {
-    this.stopEventPropagation(event);
+    // The Fob has the contenteditable attribute. This needs the event to
+    // propagate. The contenteditable attribute also stops MouseEvent from
+    // causing the strange behavior we are avoiding by stopping propagation.
```

Comment:
I would start with saying the need and then how the behavior is handled, 
"When clicking on the fob, we needs to allow MouseEvent propagate to the text span in the fob for editing. The conteneditable atrribute in the text span also calls stopPropagation() and preventDefault(), which satisfies our need to call stopEventPropagation regardless of affordance."

It would be great to have some official document to back up the claim. Yet this is the closest thing I can find https://gist.github.com/bwindels/bef8cbddea30f4b77b50
I can see stopPropagation is called on keypress

Reply:
There is no official documentation about this. The spec only has language like "This could be triggered as the default action of keydown events with various key identifiers and as the default action of mousedown events."(the spec is here https://www.w3.org/TR/2008/WD-html5-20080610/editing.html). I know empirically that it works and that is what I was going off of.

Reply:
I do not think it is correct to say that is calls stopPropagation() and preventDefault(). I believe it has similar behavior but it is not a javascript call under the hood.